routemaster/mixins/redis.rb

```diff
@@ -12,13 +11,10 @@ def self.included(by)
       end
 
       def _redis
-        @@_redis ||= ::Redis::Namespace.new(
-          ENV.fetch('ROUTEMASTER_REDIS_PREFIX'),
-          redis: ::Redis.new(url: ENV['ROUTEMASTER_REDIS_URL']))
-      end
-
-      def _redis_prefix
-        @@_redis_prefix ||= ENV.fetch('ROUTEMASTER_REDIS_PREFIX')
+        @@_redis ||=
+        ConnectionPool.wrap(size: Routemaster.config[:redis_pool_size], timeout: 2) do
```

Comment:
My understanding is that `with` avoids dynamic method dispatch using `method_missing` rather than anything to do with the connections/locking/etc. The cost of the method dispatch is likely fairly insignificant in the grand scheme of things, but given it's not a hard change to make it might be worth doing.